src/components/design_system/breadcrumbs/useBreadcrumbContext.ts

```diff
@@ -0,0 +1,23 @@
+import { trpc } from "@/client/lib/trpc";
+import { SelectableForTable } from "zapatos/schema";
+
+export type Student = SelectableForTable<"student">;
+export type Para = SelectableForTable<"user">;
+export interface BreadcrumbContext {
+  person: Student | Para | undefined;
+}
+
+export const useBreadcrumbContext = (
+  query: Record<string, string>
+): BreadcrumbContext => {
+  const { data: student } = trpc.student.getStudentById.useQuery(
```

Comment:
Firing api requests to render a breadcrumb when everything it needs is already probably somewhere in component state seems like a chatty/noisy way to do this and might under certain latencies cause a lag in breadcrumb rendering, although of course locally everything is lightning fast. Also, we'd have to add goals to this pretty soon too so we can show the goal name/title.

Reply:
Ah, my bad for not noticing that this was not introduced in this PR. Thanks @mrabbitt !